test/transport/ssh.spec.js

```diff
@@ -164,4 +163,38 @@ describe('connectSSH', () => {
     expect(options).to.have.property('username', 'admin');
     expect(options).to.not.have.property('password');
   });
+
+  it('closing with ".close()" does not emit error', () => {
+    const transport = connectSSH({
+      client,
+    });
+
+    const sshStream = new Duplex({ read: () => {} });
+    client.shell.callsArgWith(1, null, sshStream);
+    client.end.callsFake(() => {
+      sshStream.emit('end');
+    });
+    transport.on('error', errorSpy);
+
+    client.emit('ready');
+    transport.close();
+
+    expect(errorSpy).to.have.not.been.calledOnce();
+  });
+
+  it('remotely ended ssh stream emits errors', () => {
+    const transport = connectSSH({
+      client,
+    });
+
+    const sshStream = new Duplex({ read: () => {} });
+    client.shell.callsArgWith(1, null, sshStream);
+    transport.on('error', errorSpy);
+    client.emit('ready');
+
+    sshStream.emit('end');
+
+    expect(errorSpy).to.have.been.calledOnce();
+    expect(errorSpy.firstCall).to.have.been.calledWith('Connection terminated remotely');
+  });
```

Comment:
From the look of things these tests illustrate the intend we're working towards here. Client-initiated disconnects are intentional, while server-initiated ones are error-cases.